.github/workflows/tests.yml

```diff
@@ -0,0 +1,129 @@
+name: Run Tests
+
+on:
+  pull_request:
```

Comment:
Looks like we can't use a different file with one trigger because I don't see actions from `.github/workflows/checks.yml` for this PR.

Reply:
I move the checks and the tests to build-and-test.yml. As such I can make the tests depend on the compilation. 

So, order is: 
* check style and headers
* check if compilation passes + mima
* run all the tests - the tests have all independent jobs (with matrices)

Reply:
and now it seems that nothing is running 😞

Reply:
my bad, I forgot to add the new workflow file